File: app/src/main/java/org/jay/launchstarterexample/tasks/InitJPushTask.java
package org.jay.launchstarterexample.tasks;

import cn.jpush.android.api.JPushInterface;
import org.jay.launchstarter.Task;
import org.jay.launchstarterexample.MyApplication;

import java.util.ArrayList;
import java.util.List;

/**
 * 需要在getDeviceId之后执行
 */
public class InitJPushTask extends Task {

    @Override
    public List<Class<? extends Task>> dependsOn() {
        List<Class<? extends Task>> task = new ArrayList<>();
        task.add(GetDeviceIdTask.class);
        return task;
    }

    @Override
    public void run() {
        JPushInterface.init(mContext);
        MyApplication app = (MyApplication) mContext;
        JPushInterface.setAlias(mContext, 0, app.getDeviceId());
    }

}

File: app/src/main/java/org/jay/appstarterexample/tasks/InitJPushTask.kt
package org.jay.appstarterexample.tasks

import cn.jpush.android.api.JPushInterface
import org.jay.appstarter.Task
import org.jay.appstarterexample.MyApplication
import java.util.ArrayList

/**
 * 需要在getDeviceId之后执行
 */
class InitJPushTask : Task() {
    override fun dependsOn(): List<Class<out Task?>>? {
        val task: MutableList<Class<out Task?>> = ArrayList()
        task.add(GetDeviceIdTask::class.java)
        return task
    }

    override fun run() {
        JPushInterface.init(mContext)
        val app = mContext as MyApplication
        JPushInterface.setAlias(mContext, 0, app.deviceId)
    }
}